### PR TITLE
Fix extensions

### PR DESCRIPTION
**Pull request by awni**
I also enabled the extension test in Circle since a few people are using it and it will be good not to break it moving forward.

### CODE REVIEW

File: docs/src/dev/extensions.rst [hunk 1]
```diff
@@ -503,11 +503,11 @@ below.
         size_t nelem = out.size();
 
         // Encode input arrays to kernel
-        set_array_buffer(compute_encoder, x, 0);
-        set_array_buffer(compute_encoder, y, 1);
+        compute_encoder.set_input_array(x, 0);
+        compute_encoder.set_output_array(y, 1);
```
Comment (objorkman) [inline docs/src/dev/extensions.rst:507]: `set_input_array` as well to match

File: docs/src/dev/extensions.rst [hunk 2]
```diff
@@ -503,11 +503,11 @@ below.
...
 
         // Encode output arrays to kernel
-        set_array_buffer(compute_encoder, out, 2);
+        compute_encoder.set_output_buffer(out, 2);
```
Comment (objorkman) [inline docs/src/dev/extensions.rst:510]: `set_output_array` instead

File: docs/src/dev/extensions.rst (hunk 1 above)
Comment (awni) [reply docs/src/dev/extensions.rst:507]: Thanks!

**Review by angeloskath — APPROVED** [review]
Nice!

**Pull request merged**